planner/core/optimizer.go

```diff
@@ -260,6 +260,9 @@ func checkStableResultMode(sctx sessionctx.Context) bool {
 
 // DoOptimize optimizes a logical plan to a physical plan.
 func DoOptimize(ctx context.Context, sctx sessionctx.Context, flag uint64, logic LogicalPlan) (PhysicalPlan, float64, error) {
+	// TODO: move it to the logic of sync load hist-needed columns.
+	predicateColumns, _ := CollectColumnStatsUsage(logic, true, false)
+	sctx.UpdateColStatsUsage(predicateColumns)
```

Comment:
Is there any switch to control this operation?

Reply:
Yes. `tidb_enable_column_tracking` will control this operation. It will be added in next pr #30816

Reply:
Do we need to also add the switch in this pr?